schemas/README.md

```diff
@@ -10,3 +10,25 @@ The schema types for each directory are listed below.
 |---|---|---|
 | configuration/v4 | JSON Schema | https://json-schema.org |
 | messaging/v1 | Avro | https://avro.apache.org/docs/1.11.0/spec.html |
+
+### Updating the schema version
+
+Over the course of development, it is necessary to consider whether the schema version for various things should be adjusted. There are a number of files and locations which this applies to. They are as follows.
+
+- irods_environment.json _(file) (service account only)_
+- server_config.json _(file)_
+- version.json _(file)_
+- R_GRID_CONFIGURATION _(database table)_
+
+To aid with this process, the following guidelines are provided.
+
+- Keep the `schema_version` property in alignment across configuration **files**
+- Avoid schema version number conflicts across git branches
+    - Do not decrement schema version numbers
+    - Do not reuse schema version numbers
+- Understand the relationship between R_GRID_CONFIGURATION and version.json
+    - A catalog schema upgrade is triggered anytime the `catalog_schema_version` _(in version.json)_ is less than the schema version in the catalog 
+    - The current catalog schema version can be retrieved from the catalog using the following query:
+        - `select * from R_GRID_CONFIGURATION where namespace = 'database' and option_name = 'schema_version'`
+
+In general, a schema version should be adjusted if and only if there's a change in the schema's definition. This is typically reserved for major releases _(e.g. upgrading from iRODS 4.2 to iRODS 4.3)_.
```

Comment:
A few questions...

1. Should the example given refer to 4.x to 5.x? I realize that the second digit has been our major number for a long time but that's about to change, right?
2. Stupid question: Can we expand on the definition of "a change in the schema's definition"? Maybe it's obvious for other people, but not for me. :)
3. Does bumping the schema necessitate a major version release?

Reply:
> 1. Should the example given refer to 4.x to 5.x? I realize that the second digit has been our major number for a long time but that's about to change, right?

4.2 and 4.3 were used because of the branch being targeted. The words will change when I adapt this for the main branch (i.e. 4.x and 5.x may be referenced).

> 2. Stupid question: Can we expand on the definition of "a change in the schema's definition"? Maybe it's obvious for other people, but not for me. :)

Will look into that.

> 3. Does bumping the schema necessitate a major version release?

For 4.x, yes.
For 5.x, no because we get a version digit back.

Reply:
Got it, thanks

Reply:
Done